src/backend/main/py.main/caliopen_main/message/core/raw.py

```diff
@@ -35,17 +40,60 @@ def create(cls, raw):
                                              data=raw, size=size)
 
     @classmethod
+    def get(cls, raw_msg_id):
+        """
+        Get raw message from db or ObjectStorage service
+
+        :param raw_msg_id:
+        :return: a RawMessage or NotFound exception
+        """
+        try:
+            raw_msg = super(RawMessage, cls).get(raw_msg_id)
+        except NotFound:
+            raise NotFound
+
+        if raw_msg.raw_data == "" and raw_msg.uri != "":
+            # means raw message data have been stored in object store
+            # need to retrieve raw_data from it
+            url = urlparse.urlsplit(raw_msg.uri)
+            path = url.path.strip("/")
+            if url.scheme == 's3':
+                minioConf = Configuration("global").get("object_store")
+                minioClient = Minio(minioConf["endpoint"],
+                                    access_key=minioConf["access_key"],
+                                    secret_key=minioConf["secret_key"],
+                                    secure=False,
+                                    region=minioConf["raw_msg_location"])
+                try:
+                    resp = minioClient.get_object(url.netloc, path)
+                except ResponseError as exc:
+                    log.warn(exc)
+                    raise NotFound
+                # resp is a urllib3.response.HTTPResponse class
+                try:
+                    raw_msg.raw_data = resp.data
+                except Exception as exc:
```

Comment:
Any exception to a NotFound. Previously an explicit ResponseError and no others Exception catched.

It should be more consistent, and may we may raise something else than Notfound hiding the real problem (that will eventually be a warn in a log file somewhere)